### PR TITLE
Modernize CMakeLists 3

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

Update more CMakeLists.txt

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [x] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Informer/CMakeLists.txt [hunk 1]
```diff
@@ -3,20 +3,31 @@
...
 
-set(LIBRARY_SOURCES
+add_spectre_library(${LIBRARY})
+
+spectre_target_sources(
+  ${LIBRARY}
+  PRIVATE
   Informer.cpp
   Verbosity.cpp
   ${CMAKE_BINARY_DIR}/Informer/InfoAtCompile.cpp
```
Comment (fmahebert) [inline src/Informer/CMakeLists.txt:24]: The correct way to handle this source file isn't obvious to me, and also for `InfoAtLink.cpp`. Do we need a way to associate uncompiled source files with the library? Do we need to move these to a different directory? @nilsdeppe I'd like to know your thoughts on this one.

Comment (nilsdeppe) [reply src/Informer/CMakeLists.txt:24]: I think `InfoAtCompile.cpp` is handle correctly this way. Basically, we assume that `InfoAtCompile.cpp` will be generated before this is called. Maybe adding a comment that this needs to be true?

`InfoAtlink.cpp` is some very mystical think that is basically just used to inject some info late in the build process. I think we can safely ignore if for these purposes. It's unlikely to have incorrect includes and the way we track the dependency is through the header anyway. `InfoAtCompile.cpp` does need to be tracked correctly to be associated with this library.

**Review by nilsdeppe — DISMISSED** [review]

File: src/Informer/CMakeLists.txt (hunk 1 above)
Comment (fmahebert) [reply src/Informer/CMakeLists.txt:24]: Thanks @nilsdeppe for your thoughts on this. I'll update the PR tomorrow to add some comments to acknowledge (and hopefully clarify) the special status of these two sources files.

Comment (nilsdeppe) [reply src/Informer/CMakeLists.txt:24]: Okay, sounds good!

File: src/Informer/CMakeLists.txt [hunk 2]
```diff
@@ -3,20 +3,42 @@
...
+# InfoAtLink.cpp. These files contain placeholder strings that CMake fills in
+# (in copies of the files in the build directory) when configuring the build.
+#
+# We track the *generated* IntoAtCompile.cpp as a library source. This means
+# that CMake must have been configured *before* checking library dependencies
+# via the CMake function check_spectre_libs_dependencies.
+#
+# We do *not* track InfoAtLink.cpp as a library source, as it is used only to
+# inject information late in the build process. But note that InfoAtLink.cpp
+# depends on Boost.
```
Comment (fmahebert) [inline src/Informer/CMakeLists.txt:18]: @nilsdeppe Here's the new comment. Do you feel this is an accurate (and helpful?) summary of the situation? If you  have any suggestions to improve or clarify, I'm happy to implement them 🙂

Comment (nilsdeppe) [reply src/Informer/CMakeLists.txt:18]: Looks good to me!

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**